spidermon/contrib/scrapy/monitors.py

```diff
@@ -87,23 +88,60 @@ def test_should_finish_with_expected_reason(self):
 class UnwantedHTTPCodesMonitor(BaseScrapyMonitor):
     """Check for maximum number of unwanted HTTP codes.
 
-    You can configure a ``dict`` of unwanted HTTP codes with
+    Usage # 1
+
+    You can configure an ``int`` of maximum unwanted HTTP codes with
+    ``SPIDERMON_UNWANTED_HTTP_CODES_MAX_COUNT`` the default value is::
+
+        SPIDERMON_UNWANTED_HTTP_CODES_MAX_COUNT = 10
+
+        ----->
+
+        SPIDERMON_UNWANTED_HTTP_CODES = {
+            code: SPIDERMON_UNWANTED_HTTP_CODES_MAX_COUNT
+            for code in [400, 407, 429, 500, 502, 503, 504, 523, 540, 541]
+        }
+
+    Usage # 2
+
+    You can configure a ``dict`` or ``list`` of unwanted HTTP codes with
     ``SPIDERMON_UNWANTED_HTTP_CODES`` the default value is::
 
-        DEFAULT_ERROR_CODES = {
+        1.1 - In case of Dictionary:
+
+        SPIDERMON_UNWANTED_HTTP_CODES = {
             code: 10
-            for code in [400, 407, 429, 500, 502, 503, 504, 523, 540, 541]}
-    """
+            for code in [400, 407, 429, 500, 502, 503, 504, 523, 540, 541]
+        }
+
+        1.2 - In case of List:
 
-    DEFAULT_ERROR_CODES = {
-        code: 10 for code in [400, 407, 429, 500, 502, 503, 504, 523, 540, 541]
-    }
+        SPIDERMON_UNWANTED_HTTP_CODES = [400, 407, 429, 500, 502, 503, 504, 523, 540, 541]
+
+        ----->
+
+        SPIDERMON_UNWANTED_HTTP_CODES = {
+            code: SPIDERMON_UNWANTED_HTTP_CODES_MAX_COUNT / 10
+            for code in [400, 407, 429, 500, 502, 503, 504, 523, 540, 541]
+        }
+
+    """
+    DEFAULT_ERROR_MAX_COUNT = 10
+    DEFAULT_ERROR_CODES = [400, 407, 429, 500, 502, 503, 504, 523, 540, 541]
 
     @monitors.name("Should not hit the limit of unwanted http status")
     def test_check_unwanted_http_codes(self):
-        error_codes = self.crawler.settings.getdict(
+        error_codes = self.crawler.settings.get(
             SPIDERMON_UNWANTED_HTTP_CODES, self.DEFAULT_ERROR_CODES
         )
+
+        errors_max_count = self.crawler.settings.getint(
+            SPIDERMON_UNWANTED_HTTP_CODES_MAX_COUNT, self.DEFAULT_ERROR_MAX_COUNT
+        )
+
+        if not isinstance(error_codes, dict):
+            error_codes = {code: errors_max_count for code in error_codes}
+
         for code, max_errors in error_codes.items():
             code = int(code)
```

Comment:
I believe that the codes are always integers (make this clear in the docs, BTW), so we don't need to convert it here.

Reply:
@rennerocha I have not touched this part of code. It was there pre hand.

Reply:
@rennerocha If it is sent through command line as @Gallaecio mentioned above then it will be str

Reply:
Actually, once you load them with `json.loads`, they should be `int`.

Reply:
@Gallaecio What if it is a list and we split the value based on 
``return value.split(',')``
Would we need that int(code) then ?

Reply:
You are completely right, I had forgotten about comma-separated lists being supported as well!